app/models/question.rb

```diff
@@ -201,18 +201,20 @@ def annotations_per_org(org_id)
   # upon saving of question update conditions (via a delete and create) from params
   # the old_to_new_opts map allows us to rewrite the question_option ids which may be out of sync
   # after versioning
-  def update_conditions(param_conditions, old_to_new_opts, question_id_map)
+  def update_conditions(param_conditions, flash, old_to_new_opts, question_id_map)
+    res = true
     self.conditions.destroy_all
 
     if param_conditions.present?
       param_conditions.each do |_key, value|
-        saveCondition(value, old_to_new_opts, question_id_map)
+        res = saveCondition(value, flash, old_to_new_opts, question_id_map)
```

Comment:
`res` will always be whatever the result of the last condition are, so if there are 3 and the 2nd one is false this will still return true.
Maybe add a `break if res == false`?